src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1486 @@
+                                   Vec<OsIpcSharedMemory>),WinError> {
+        let mut receiver = self.receiver;
+        try!(receiver.accept());
+        let (data, channels, shmems) = try!(receiver.recv());
+        Ok((receiver, data, channels, shmems))
+    }
+}
+
+pub enum OsIpcChannel {
+    Sender(OsIpcSender),
+    Receiver(OsIpcReceiver),
+}
+
+#[derive(PartialEq, Debug)]
+pub struct OsOpaqueIpcChannel {
+    handle: HANDLE,
+}
+
+impl OsOpaqueIpcChannel {
+    fn new(handle: HANDLE) -> OsOpaqueIpcChannel {
+        OsOpaqueIpcChannel {
+            handle: handle,
+        }
+    }
+
+    pub fn to_receiver(&mut self) -> OsIpcReceiver {
+        unsafe { OsIpcReceiver::from_handle(self.handle) }
+    }
+
+    pub fn to_sender(&mut self) -> OsIpcSender {
+        unsafe { OsIpcSender::from_handle(self.handle) }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum WinError {
+    WindowsResult(u32),
+    ChannelClosed,
+    NoData,
+}
+
+impl WinError {
+    pub fn error_string(errnum: u32) -> String {
+        // This value is calculated from the macro
+        // MAKELANGID(LANG_SYSTEM_DEFAULT, SUBLANG_SYS_DEFAULT)
+        let lang_id = 0x0800 as winapi::DWORD;
+        let mut buf = [0 as winapi::WCHAR; 2048];
+
+        unsafe {
+            let res = kernel32::FormatMessageW(winapi::FORMAT_MESSAGE_FROM_SYSTEM |
+                                               winapi::FORMAT_MESSAGE_IGNORE_INSERTS,
+                                               ptr::null_mut(),
+                                               errnum as winapi::DWORD,
+                                               lang_id,
+                                               buf.as_mut_ptr(),
+                                               buf.len() as winapi::DWORD,
+                                               ptr::null_mut()) as usize;
+            if res == 0 {
+                // Sometimes FormatMessageW can fail e.g. system doesn't like lang_id,
+                let fm_err = kernel32::GetLastError();
+                return format!("OS Error {} (FormatMessageW() returned error {})",
+                               errnum, fm_err);
+            }
+
+            match String::from_utf16(&buf[..res]) {
+                Ok(msg) => {
+                    // Trim trailing CRLF inserted by FormatMessageW
+                    msg.trim().to_string()
+                },
+                Err(..) => format!("OS Error {} (FormatMessageW() returned \
+                                    invalid UTF-16)", errnum),
+            }
+        }
+    }
+
+    fn from_system(err: u32, f: &str) -> WinError {
+        win32_trace!("WinError: {} ({}) from {}", WinError::error_string(err), err, f);
+        WinError::WindowsResult(err)
+    }
+
+    fn last(f: &str) -> WinError {
+        WinError::from_system(GetLastError(), f)
+    }
+
+    pub fn channel_is_closed(&self) -> bool {
+        match *self {
+            WinError::ChannelClosed => true,
+            _ => false,
+        }
```

Comment:
If you derive `PartailEq` on `WinError`, you could just use an ordinary comparison here, like in the `inprocess` back-end.